src/gd.c

```diff
@@ -895,19 +895,24 @@ BGD_DECLARE(void) gdImageColorDeallocate (gdImagePtr im, int color)
  */
 BGD_DECLARE(void) gdImageColorTransparent (gdImagePtr im, int color)
 {
-	if (color < 0) {
+	// Reset ::transparent
+	if (color == -1) {
+		im->transparent = -1;
+	}
+	if (color < -1) {
+		return;
+	}
+	if (im->trueColor) {
+		im->transparent = color;
 		return;
 	}
```

Comment:
This looks superfluous; at least previously, we didn't do anything to truecolor images.

Reply:
For clarity and simplify paths. Can be removed but it provides clarity and clear return here. I prefer that that what we had before. Thoughts?